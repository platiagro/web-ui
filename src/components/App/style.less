.main-layout {
  height: 100vh;
  overflow: hidden;
}
