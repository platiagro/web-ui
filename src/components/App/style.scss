@import '../../_variables.scss';

h1 {
  color: $primaryColor;
  text-align: center;
}
