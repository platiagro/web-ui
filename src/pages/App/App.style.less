// MARGIN BOTTOM NEW BUTTONS
@margin-bottom-buttons: 20px;

.main-layout {
  height: 100vh;
  overflow: hidden;
}

// CONTENT PAGE
.contentPage {
  margin: 20px 20px 0px;
  padding: 10px;
}

// NEW PROJECT BUTTON
.newProjectButton {
  margin-bottom: @margin-bottom-buttons;
}

// NEW PROJECT BUTTON
.newTaskButton {
  margin-bottom: @margin-bottom-buttons;
}
