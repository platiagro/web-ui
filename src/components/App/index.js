// CORE LIBS
import React from 'react';

// UI LIBS
import { Layout } from 'antd';

// COMPONENTS
import MainSider from '../MainSider/_';
import Content from '../Content/_';

import './style.less';

/**
 * Main component of the application.
 *
 * This component is responsible for structuring the main layout of the
 * application.
 *
 * This component is also responsible for routing the main content of the
 * application.
 */
const App = () => (
  // layout container
  <Layout className='main-layout'>
    {/* main sider */}
    <MainSider />
    {/* content */}
    <Content />
  </Layout>
);

// EXPORT
export default App;
