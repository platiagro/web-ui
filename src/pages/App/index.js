import React from 'react';
import { Layout } from 'antd';

import { MainRoutes } from 'routes';
import MainSider from 'components/MainSider/_';

import './style.less';

const App = () => (
  <Layout className='main-layout'>
    <MainSider />

    <Layout>
      <MainRoutes />
    </Layout>
  </Layout>
);

export default App;
