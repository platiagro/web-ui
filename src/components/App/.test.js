import React from 'react';

import { shallow } from 'enzyme';

import App from '.';

describe('App component should', () => {
  it('renders without crashing', () => {
    const location = {
      pathname: '/',
    };

    shallow(<App location={location} />);
  });

  it('renders html correctly with valid route', () => {
    const location = {
      pathname: '/',
    };

    const appShalowed = shallow(<App location={location} />);

    expect(appShalowed).toMatchSnapshot();
  });

  it('renders html correctly with invalid route', () => {
    const location = {
      pathname: '/xyz-invalid-url-path',
    };

    const appShalowed = shallow(<App location={location} />);

    expect(appShalowed).toMatchSnapshot();
  });
});
