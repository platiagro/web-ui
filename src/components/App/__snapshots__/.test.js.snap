// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component is expected render html correctly 1`] = `
<Adapter>
  <Adapter>
    <Adapter>
      <Route
        render={[Function]}
      />
    </Adapter>
    <Adapter>
      <Switch>
        <Route
          component={[Function]}
          exact={true}
          key="/"
          path="/"
        />
        <Route
          component={[Function]}
          exact={true}
          key="/projects"
          path="/projects"
        />
        <Route
          component={[Function]}
          key="/projects/:projectId/:experimentId?"
          path="/projects/:projectId/:experimentId?"
        />
        <Route
          component={[Function]}
          key="/implanted-flows"
          path="/implanted-flows"
        />
        <Route
          component={[Function]}
          key="*"
          path="*"
        />
      </Switch>
    </Adapter>
    <Adapter>
      <MainFooter />
    </Adapter>
  </Adapter>
</Adapter>
`;
